types/SocketEvent.js

```diff
@@ -29,5 +29,4 @@ export default {
   UPDATED: 'updated',
   REMATCH: 'rematch',
   NEW_GAME_FOR_REMATCH: 'newGameForRematch',
-  JOIN_REMATCH: 'joinRematch',
```

Comment:
This isn't necessary as a socket event. It represents when a player joins a rematch game after it's created i.e. subscribes to it for updates, but there is nothing that other users need to do to respond to the event, so we don't need it.

It was previously used to handle spectators subscribing to the game, but instead they now respond to the `NEW_GAME_FOR_REMATCH` event just like the players do